pkg/kapp/resources/resources.go

```diff
@@ -401,6 +401,18 @@ func (c *ResourcesImpl) Exists(resource Resource) (bool, error) {
 			return isDone, c.resourceErr(err, "Checking existence of", resource)
 		}
 
+		if existsOpts.checkForSameUID() {
+				// If fetchedRes(i.e. resource from K8s) is not null and its UID didn't match with the
+				// UID of resource we are trying to delete, then it means resource has been deleted
+				// successfully.
+				if fetchedRes != nil {
+					if string(fetchedRes.GetUID()) != resource.UID() {
```

Comment:
we should be able to use .UID() instead of GetUID()

Reply:
As it is of type Unstructured struct from k8s/apimachinery, it does have UID() function. It has SetUID() and GetUID() functions. https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured